File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> m;
        for(int i=0; i<nums.size(); i++)
            m[nums[i]] = -1;
        
        for(int i=0; i<nums.size(); i++){
            if(m[nums[i]]>=0)
                if(abs(i-m[nums[i]])<=k) return true;
            m[nums[i]] = i;
        }
        return false;
    }
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.go
func containsNearbyDuplicate(nums []int, k int) bool {
    m := make(map[int]int)
    for i, num := range nums {
        if val, ok := m[num]; ok {
            if i-val <= k {
                return true
            }
        }
        m[num] = i
    }
    return false
}